WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModelTests.swift

```diff
@@ -0,0 +1,307 @@
+import XCTest
+import Yosemite
+@testable import WooCommerce
+import protocol Storage.StorageManagerType
+import protocol Storage.StorageType
+
+final class ReviewsDashboardCardViewModelTests: XCTestCase {
+    private let sampleSiteID: Int64 = 1337
+    private var stores: MockStoresManager!
+
+    private let sampleReviews: [ProductReview] = [ProductReview.fake().copy(siteID: 1337, reviewID: 1),
+                                 ProductReview.fake().copy(siteID: 1337, reviewID: 2),
+                                 ProductReview.fake().copy(siteID: 1337, reviewID: 3)]
+
+    /// Mock Storage: InMemory
+    private var storageManager: StorageManagerType!
+
+    /// View storage for tests
+    private var storage: StorageType {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        super.setUp()
+        storageManager = MockStorageManager()
+        stores = MockStoresManager(sessionManager: SessionManager.makeForTesting())
+    }
+
+    override func tearDown() {
+        stores = nil
+        storageManager = nil
+        super.tearDown()
+    }
+
+    @MainActor
+    func test_reviews_are_loaded_from_storage_when_available() async {
+        // Given
+        let viewModel = ReviewsDashboardCardViewModel(siteID: sampleSiteID,
+                                                      stores: stores,
+                                                      storageManager: storageManager)
+        insertReviews(sampleReviews)
+
+        // When
+        stores.whenReceivingAction(ofType: ProductReviewAction.self) { action in
+            switch action {
+            case let .synchronizeProductReviews(_, _, _, _, _, onCompletion):
+                onCompletion(.success(self.sampleReviews))
+            default:
+                XCTFail("Unexpected action: \(action)")
+            }
+        }
+
+        stores.whenReceivingAction(ofType: ProductAction.self) { action in
+            switch action {
+            case let .retrieveProducts(_, _, _, _, onCompletion):
+                onCompletion(.success(([], true)))
+            default:
+                XCTFail("Unexpected action: \(action)")
+            }
+        }
+
+        stores.whenReceivingAction(ofType: NotificationAction.self) { action in
+            switch action {
+            case let .synchronizeNotifications(onCompletion):
+                onCompletion(nil)
+            default:
+                XCTFail("Unexpected action: \(action)")
+            }
+        }
+
+        await viewModel.reloadData()
+
+        // Then
+        XCTAssertEqual(viewModel.data.count, self.sampleReviews.count)
```

Comment:
We should compare the data rather than just the count, to make sure that the details are correct.

Reply:
Thanks, updated in https://github.com/woocommerce/woocommerce-ios/pull/12806/commits/8739e01865f91f53a138a6f3aa2bd12f9de5597f